market-generator/src/main/java/com/market/banica/generator/configuration/MarketConfigurationImpl.java

```diff
@@ -15,183 +16,191 @@
-            origin = origin.toLowerCase(Locale.getDefault());
             good = good.toLowerCase(Locale.getDefault());
+            if (!this.doesGoodExist(good)) {
+                throw new IllegalArgumentException(String
+                        .format("A good with name %s does not exist and cannot be updated", good.toUpperCase()));
+            }
 
-            Properties properties = new LinkedProperties();
-            boolean append = false;
-            boolean condition = !this.doesGoodExist(String.format(ORIGIN_GOOD_PATTERN,
-                    origin, good));
-
-
-            this.modifyProperty(origin, good,
-                    new NotFoundException(errorMessage),
-                    condition,
+            GoodSpecification updatedGoodSpecification = new GoodSpecification(good,
                     quantityLow, quantityHigh, quantityStep,
                     priceLow, priceHigh, priceStep,
-                    periodLow, periodHigh, periodStep, append, properties, properties::setProperty, loggerMessage);
+                    periodLow, periodHigh, periodStep);
+
+            this.modifyProperty(updatedGoodSpecification);
+
+            LOGGER.info("Updating an existing goodSpecification.");
         } finally {
-            this.lock.writeLock().unlock();
+            propertiesWriteLock.writeLock().unlock();
         }
     }
 
-    private void modifyProperty(String origin, String good,
-                                RuntimeException exception, boolean condition,
-                                long quantityLow, long quantityHigh, long quantityStep,
-                                double priceLow, double priceHigh, double priceStep,
-                                int periodLow, int periodHigh, int periodStep, boolean append,
-                                Properties properties, BiConsumer<String, String> function, String loggerMessage) {
-
-        if (condition) {
-            throw exception;
-        }
+    private void modifyProperty(GoodSpecification modifiedGoodSpecification) {
 
-        GoodSpecification goodSpecification = new GoodSpecification(good,
-                quantityLow, quantityHigh, quantityStep,
-                priceLow, priceHigh, priceStep,
-                periodLow, periodHigh, periodStep);
+        validateGoodSpecification(modifiedGoodSpecification);
 
-        validateGoodSpecification(goodSpecification);
+        modifiedGoodSpecification.generateProperties().forEach(properties::setProperty);
 
-        this.modifyProperties(goodSpecification.generateProperties(origin)
-                , properties, function, this.file, append);
+        this.saveProperties();
 
-        this.goods.put(String.format(ORIGIN_GOOD_PATTERN, origin, good), goodSpecification);
+        this.goods.put(modifiedGoodSpecification.getName(), modifiedGoodSpecification);
 
-        LOGGER.info(loggerMessage);
     }
 
     private void validateGoodSpecification(GoodSpecification goodSpecification) {
```

Comment:
Shouldn't we  add a check for negative price here?

Reply:
I must have missed the negative validation, added validation and warning logs to all.